scaffoldmaker/meshtypes/meshtype_3d_lens1.py

```diff
@@ -0,0 +1,250 @@
+"""
+Generates a 3-D lens mesh with variable numbers of elements
+around, up the central axis, and radially.
+"""
+
+from __future__ import division
+import math
+from scaffoldmaker.meshtypes.meshtype_3d_solidsphere1 import MeshType_3d_solidsphere1
+from scaffoldmaker.utils.meshrefinement import MeshRefinement
+from scaffoldmaker.utils.zinc_utils import *
+from opencmiss.zinc.element import Element, Elementbasis, Elementfieldtemplate
+from opencmiss.zinc.field import Field
+from opencmiss.zinc.node import Node
+
+class MeshType_3d_lens1:
+    '''
+    classdocs
+    '''
+    @staticmethod
+    def getName():
+        return '3D Lens 1'
+
+    @staticmethod
+    def getDefaultOptions():
+        options = MeshType_3d_solidsphere1.getDefaultOptions()
+        options['Number of elements around'] = 8
+        optionsLens = {
+            'Axial thickness' : 4.0,
+            'Anterior radius of curvature' : 10.0,
+            'Posterior radius of curvature' : 6.0,
+            'Spherical radius fraction' : 0.7
+        }
+        options.update(optionsLens)
+        return options
+
+    @staticmethod
+    def getOrderedOptionNames():
+        optionNames = MeshType_3d_solidsphere1.getOrderedOptionNames()
+        optionNames.remove('Diameter')
+        for optionName in [
+            'Spherical radius fraction',
+            'Posterior radius of curvature',
+            'Anterior radius of curvature',
+            'Axial thickness']:
+            optionNames.insert(3, optionName)
+        return optionNames
+
+    @staticmethod
+    def checkOptions(options):
+        MeshType_3d_solidsphere1.checkOptions(options)
+        for key in [
+            'Spherical radius fraction',
+            'Posterior radius of curvature',
+            'Anterior radius of curvature',
+            'Axial thickness']:
+            if options[key] < 0.0:
+                options[key] = 0.0
+        if options['Spherical radius fraction'] > 1.0:
+            options['Spherical radius fraction'] = 1.0
+        if options['Posterior radius of curvature'] < options['Axial thickness']*0.5:
+            options['Posterior radius of curvature'] = options['Axial thickness']*0.5
+        if options['Anterior radius of curvature'] < options['Axial thickness']*0.5:
+            options['Anterior radius of curvature'] = options['Axial thickness']*0.5
+
+    @classmethod
+    def sphereToLens(cls, region, options):
```

Comment:
Pass actual parameters to this function, not options dict. I.e.. coordinates, radiusSphere, radiusAnt, radiusPos, lensThickness, sphericalRadiusFraction etc. Describe each argument using :param NAME: description.

Can get fieldmodule from coordinates field, so no need to pass:
fm = coordinates.getFieldmodule()

Function and description should state that it defines a field giving positions on a spherical lens as functions of positions in the sphere.

I would make this a standalone function e.g.:
def getSphereToLensCoordinates(args):